### PR TITLE
Prepare configuration necessary for talking to IDAM and CCD

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-810

### Change description ###

Prepare configuration necessary for talking to IDAM and CCD, so that this application can update exception records in CCD.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/bulk-scan-payment-processor/values.yaml [hunk 1]
```diff
@@ -7,6 +7,8 @@ java:
     S2S_NAME: "bulk_scan_payment_processor"
     S2S_URL: http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
     PAY_HUB_URL: http://ccpay-bulkscanning-api-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
+    IDAM_API_URL: https://idam-api.{{ .Values.global.environment }}.platform.hmcts.net
+    IDAM_CLIENT_REDIRECT_URI: https://bulk-scan-orchestrator-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal/oauth2/callback
```
Comment (nitinprabhu) [inline charts/bulk-scan-payment-processor/values.yaml]: These envs would need to overrided in preview yaml to use AAT ones ?

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -45,6 +45,16 @@ idam:
...
+  api:
+    url: ${IDAM_API_URL}
+  client:
+    id: 'bsp'
+    secret: ${IDAM_CLIENT_SECRET}
+    redirect_uri: ${IDAM_CLIENT_REDIRECT_URI}
+  users:
+    bulkscan:
+      username: ${IDAM_USERS_BULKSCAN_USERNAME:bulkscan+ccd@gmail.com}
+      password: ${IDAM_USERS_BULKSCAN_PASSWORD:Password12}
```
Comment (nitinprabhu) [inline src/main/resources/application.yaml]: If we can remove default that helps reason is if you don't set up env vars in any particular environment application boot fails and it is easy to isolate the issue other wise default would be set and might take a while when tests fails.

File: charts/bulk-scan-payment-processor/values.yaml [hunk 3]
```diff
@@ -18,6 +20,9 @@ java:
         - site-id-finrem
         - payments-queue-listen-connection-string
         - app-insights-instrumentation-key
+        - idam-client-secret
+        - idam-users-bulkscan-username
```
Comment (nitinprabhu) [inline charts/bulk-scan-payment-processor/values.yaml]: I would imagine you would add idam user config for other jurisdiction later e.g probate

File: charts/bulk-scan-payment-processor/values.yaml (hunk 1 above)
Comment (lgonczar) [reply charts/bulk-scan-payment-processor/values.yaml]: So `{{ .Values.global.environment }}` won't resolve to `aat` in aat?

File: src/main/resources/application.yaml (hunk 2 above)
Comment (lgonczar) [reply src/main/resources/application.yaml]: That's the approach I started with (didn't include any user), but got some mapping errors at application startup. I'll check what happens if I remove `users` root element from here - previously I had it included (and empty).

File: charts/bulk-scan-payment-processor/values.yaml (hunk 3 above)
Comment (lgonczar) [reply charts/bulk-scan-payment-processor/values.yaml]: Yes, I'll add other service users as the last step in this story

Comment (lgonczar) [reply charts/bulk-scan-payment-processor/values.yaml]: On a second thought, adding them now would help avoid bumping version twice. Will do it now.

**Pull request merged**